src/query-builder/aggregate-function-builder.ts

```diff
@@ -0,0 +1,114 @@
+import { freeze } from '../util/object-utils.js'
+import { AggregateFunctionNode } from '../operation-node/aggregate-function-node.js'
+import { OperationNodeSource } from '../operation-node/operation-node-source.js'
+import { AliasNode } from '../operation-node/alias-node.js'
+import { IdentifierNode } from '../operation-node/identifier-node.js'
+import { preventAwait } from '../util/prevent-await.js'
+import { OverBuilder } from './over-builder.js'
+import { createOverBuilder } from '../parser/parse-utils.js'
+
+export class AggregateFunctionBuilder<DB, TB extends keyof DB, O = unknown>
+  implements OperationNodeSource
+{
+  readonly #props: AggregateFunctionBuilderProps
+
+  constructor(props: AggregateFunctionBuilderProps) {
+    this.#props = freeze(props)
+  }
+
+  /**
+   * TODO: ...
+   */
+  as<A extends string>(
+    alias: A
+  ): AliasedAggregateFunctionBuilder<DB, TB, O, A> {
+    return new AliasedAggregateFunctionBuilder(this, alias)
+  }
+
+  /**
+   * TODO: ...
+   */
+  distinct(): AggregateFunctionBuilder<DB, TB, O> {
+    return new AggregateFunctionBuilder({
+      ...this.#props,
+      aggregateFunctionNode: AggregateFunctionNode.cloneWithDistinct(
+        this.#props.aggregateFunctionNode
+      ),
+    })
+  }
+
+  /**
+   * TODO: ...
+   */
+  over(
```

Comment:
Does `over()` without any arguments produce meaningful SQL?

Reply:
This use case is mentioned in official documentations ([postgres](https://www.postgresql.org/docs/current/tutorial-window.html), [mysql](https://dev.mysql.com/doc/refman/8.0/en/window-functions-usage.html)).

Basically "let aggregate function do its thing, but duplicate it across all rows and return all rows instead of a single row".

> The first OVER clause is empty, which treats the entire set of query rows as a single partition. The window function thus produces a global sum, but does so for each row.